tests/test_flow/test_persistence_random.py

```diff
@@ -338,15 +381,34 @@ def _random_existing_entity_name(self):
         return self._rng.choice(self._harness.get_all_entity_names())
 
     def _random_descriptor_of_new_entities(self, should_persist):
-        def new():
+        def new_entity_name():
             entity_name = self._harness.create_entity(should_persist)
             self._child_counts_by_entity_name[entity_name] = 0
             return entity_name
 
-        if self._random_bool(p_true=1 - self._config.p_multi_out):
-            return new()
+        include_multiple_entities = self._random_bool(p_true=self._config.p_multi_out)
+        if not include_multiple_entities:
+            return new_entity_name()
+
+        include_three_entities = self._random_bool(
+            p_true=self._config.p_three_out_given_multi
+        )
+        if not include_three_entities:
+            template = "X, X"
+
         else:
-            return f"{new()}, {new()}"
+            # We'll randomly choose one of the following three-entity descriptors:
+            template = self._rng.choice(
+                [
+                    "X, X, X",
```

Comment:
This is so much better!

Reply:
I agree! Thanks for pushing back on the original version.